src/WalkingModule/src/Module.cpp

```diff
@@ -499,10 +509,17 @@ bool WalkingModule::updateModule()
                 return false;
             }
 
-            m_retargetingClient->reset(m_FKSolver->getHeadToWorldTransform().inverse()
-                                       * m_FKSolver->getLeftHandToWorldTransform(),
-                                       m_FKSolver->getHeadToWorldTransform().inverse()
-                                       * m_FKSolver->getRightHandToWorldTransform());
+            if(!m_retargetingClient->reset(m_FKSolver->getHeadToWorldTransform().inverse()
+                                          * m_FKSolver->getLeftHandToWorldTransform(),
+                                          m_FKSolver->getHeadToWorldTransform().inverse()
+                                          * m_FKSolver->getRightHandToWorldTransform(),
+                                          m_robotControlHelper->getJointPosition(),
```

Comment:
Can you get also the joint values from the ``m_FKSolver`` to avoid the problem we had about the discrepancy? In principle, you could pass directly the ``m_FKSolver`` object.

Reply:
> Can you get also the joint values from the `m_FKSolver` to avoid the problem we had about the discrepancy?

Yes, actually you spot the bug already discovered by @MiladShafiee.  I'm updating the code

> In principle, you could pass directly the `m_FKSolver` object.

Yes it would be nice, but I would avoid to do it within this PR

Reply:
> Yes, actually you spot the bug already discovered by @MiladShafiee. I'm updating the code

Actually no. The bug is in 
```cpp
 ok &= solver->setRobotState(m_robotControlHelper->getJointPosition(),                            
                                 m_FKSolver->getLeftFootToWorldTransform(),
                                 m_FKSolver->getRightFootToWorldTransform(),
                                 m_FKSolver->getLeftHandToWorldTransform(),
                                 m_FKSolver->getRightHandToWorldTransform(),
                                 m_FKSolver->getNeckOrientation(),
                                 m_FKSolver->getCoMPosition());
```
The one you point is correct. The `retargetingClient` required the current position of the joints

Reply:
Actually, I suggested to pass the joint positions from the ``FKSolver`` to avoid exactly this kind of problems. Otherwise, how can I understand if this method needs the measured or the desired values? It seems very confusing.

Reply:
Great! Can you pass it as a ``const`` object or there are issues? Can the same be done also for the ``setRobotState``? (Just wondering)

Reply:
> Can the same be done also for the setRobotState? (Just wondering)

Here you have: ab9da12a431fbf7b77bd0fce5ae2d47d82556525ab9da12a431fbf7b77bd0fce5ae2d47d82556525